Src/FluentAssertions.Json/JTokenAssertions.cs

```diff
@@ -84,13 +85,36 @@ public JTokenAssertions(JToken subject) : base(subject)
         public AndConstraint<JTokenAssertions> BeEquivalentTo(JToken expected, string because = "",
             params object[] becauseArgs)
         {
-            return BeEquivalentTo(expected, false, because, becauseArgs);
+            return BeEquivalentTo(expected, false,null, because, becauseArgs);
         }
 
-        private AndConstraint<JTokenAssertions> BeEquivalentTo(JToken expected, bool ignoreExtraProperties, string because = "",
+        /// <summary>
+        ///     Asserts that the current <see cref="JToken" /> is equivalent to the <paramref name="expected" /> element,
+        ///     using an equivalent of <see cref="JToken.DeepEquals(JToken, JToken)" />.
+        /// </summary>
+        /// <param name="expected">The expected element</param>
+        /// <param name="config">The options to consider while asserting values</param>
+        /// <param name="because">
+        ///     A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        ///     is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        ///     Zero or more objects to format using the placeholders in <see paramref="because" />.
+        /// </param>
+        public AndConstraint<JTokenAssertions> BeEquivalentTo(JToken expected,
+            Func<EquivalencyAssertionOptions<object>, EquivalencyAssertionOptions<object>> config,
```

Comment:
🤔 I'm not fully convinced yet that directly exposing `EquivalencyAssertionOptions` on the public API of this method is the way to go. Although this class is `public` in FluentAssertions and breaking changes are controlled well, a lot of the methods and properties on that type will not be relevant for JSON comparisons. So I'm wondering whether it would be wise to wrap `EquivalencyAssertionOptions` in a custom type that only exposes the relevant members. What do you think @jnyrup ?

Reply:
`EquivalencyAssertionOptions` can surely be overwhelming.
So the test writer experience would be improved if we only offered "the relevant" subset of options.
Contrary it could give some confusion why some option is available in regular FluentAssertions, but not in FluentAssertions.Json.

I haven't used FluentAssertions.Json enough to tell what properties are relevant or not.
People use FluentAssertions in so many ways, that every/most option we leave out here would have a use case to someone.

Reply:
But a lot of them are not applicable. So I'd rather have it use a different type of `EquivalencyAssertionOptions`, or an interface that only exposes the relevant parts. Then we can gradually add more if that is needed.

Reply:
> But a lot of them are not applicable. So I'd rather have it use a different type of EquivalencyAssertionOptions, or an interface that only exposes the relevant parts. Then we can gradually add more if that is needed.

I agree, I'll try add a new type of `EquivalencyAssertionOptions`.

Reply:
Or maybe an interface is good enough. Then you can inherit the underlying implementation from the main `EquivalencyAssertionOptions`.